src/serializer/ResidualReactElementSerializer.js

```diff
@@ -156,7 +160,10 @@ export class ResidualReactElementSerializer {
           originalCreateElementIdentifier = this.residualHeapSerializer.serializeValue(createElement);
 
           if (shouldHoist) {
-            const optimizedFunction = this.residualHeapSerializer.tryGetOptimizedFunctionRoot(value);
+            const optimizedFunction = tryGetOptimizedFunctionRoot(
```

Comment:
Instead of this pattern where we create helper objects (helper state) all the time to pass it to some static function, create a dedicated class that encapsulates the data and code. Maybe `ResidualOptimizedFunctions`.